src/lib/parsers/Args.ts

```diff
@@ -440,11 +441,77 @@ export class Args {
 		throw result.error;
 	}
 
+	/**
+	 * Retrieves the next raw argument from the parser.
+	 * @example
+	 * ```typescript
+	 * // !numbers 1 2 3
+	 *
+	 * console.log(args.nextMaybe());
+	 * // -> { exists: true, value: '1' }
+	 * ```
+	 */
+	public nextMaybe(): Maybe<string>;
+	/**
+	 * Retrieves the value of the next unused ordered token, but only if it could be transformed.
+	 * That token will now be consider used if the transformation succeeds.
+	 * @typeparam T Output type of the [[ArgsNextCallback callback]].
+	 * @param cb Gives an option of either the resulting value, or nothing if failed.
+	 * @example
+	 * ```typescript
+	 * // !numbers 1 2 3
+	 * const parse = (x: string) => {
+	 *   const n = Number(x);
+	 *   return Number.isNaN(n) ? none() : some(n);
+	 * };
+	 *
+	 * console.log(args.nextMaybe(parse));
+	 * // -> { exists: true, value: 1 }
+	 * ```
+	 */
+	public nextMaybe<T>(cb: ArgsNextCallback<T>): Maybe<T>;
+	public nextMaybe<T>(cb?: ArgsNextCallback<T>): Maybe<T | string> {
+		return maybe<T | string>(typeof cb === 'function' ? this.parser.singleMap(cb) : this.parser.single());
+	}
+
+	/**
+	 * Similar to [[Args.nextMaybe]] but returns the value on success, null otherwise.
+	 * @example
+	 * ```typescript
+	 * // !numbers 1 2 3
+	 *
+	 * console.log(args.next());
+	 * // -> '1'
+	 * ```
+	 */
+	public next(): string;
+	/**
+	 * Similar to [[Args.nextMaybe]] but returns the value on success, null otherwise.
+	 * @typeparam T Output type of the [[ArgsNextCallback callback]].
+	 * @param cb Gives an option of either the resulting value, or nothing if failed.
+	 * @example
+	 * ```typescript
+	 * // !numbers 1 2 3
+	 * const parse = (x: string) => {
+	 *   const n = Number(x);
+	 *   return Number.isNaN(n) ? none() : some(n);
+	 * };
+	 *
+	 * console.log(args.nextMaybe(parse));
+	 * // -> 1
+	 * ```
+	 */
+	public next<T>(cb: ArgsNextCallback<T>): T;
+	public next<T>(cb?: ArgsNextCallback<T>): T | string | null {
+		const value = cb ? this.nextMaybe(cb) : this.nextMaybe();
```

Comment:
No need for ternary

Reply:
There is a need for ternary, they're different overloads with different returns.

Believe it or not, TypeScript hasn't figured out how to `|` the return type when you call multiple overloads in 8 years.